src/Symfony/Component/Cache/Adapter/ChainAdapter.php

```diff
@@ -145,12 +145,12 @@ public function getItem($key)
     /**
      * {@inheritdoc}
      */
-    public function getItems(array $keys = [])
+    public function getItems(array $keys = []): \Traversable
```

Comment:
Our own `AdapterInterface` was more strict in its `@return` annotation and (apart from one broken mock) all of our implementations adhered to that. That's why I went with `Traversable` here. Note that `AdapterInterface` is also narrowing down `getItem()` by declaring that *our* `CacheItem` is returned and not just *any* `CacheItemInterface` implementation.

Being stricter here could make the maintenance easier for us because when calling our own adapters we can be sure that we don't have to handle the edge case of an array return value. But then again, our current code would still work fine if one of our adapters starts to return arrays, as far as I can tell.

Reply:
> AdapterInterface is also narrowing down getItem() by declaring that our CacheItem is returned

correct, that's by design in PSR-6

> our current code would still work fine if one of our adapters would suddenly start to return arrays

then we have no reason to restrict here IMHO